packager/util/util.go

```diff
@@ -76,3 +78,32 @@ func CanSignHttpExchanges(cert *x509.Certificate) bool {
 	}
 	return false
 }
+
+func CheckCertificate(cert *x509.Certificate, priv crypto.PrivateKey, domain string) error {
+	certPubKey := cert.PublicKey.(*ecdsa.PublicKey)
+	pubKey := priv.(*ecdsa.PrivateKey).PublicKey
+	if certPubKey.Curve != pubKey.Curve {
+		return errors.New("PublicKey.Curve not match")
+	}
+	if certPubKey.X.Cmp(pubKey.X) != 0 {
+		return errors.New("PublicKey.X not match")
+	}
+	if certPubKey.Y.Cmp(pubKey.Y) != 0 {
+		return errors.New("PublicKey.Y not match")
+	}
+	if err := cert.VerifyHostname(domain); err != nil {
+		return err
+	}
+	// https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#cross-origin-cert-req
+	// Clients MUST reject certificates with this extension that were issued after 2019-05-01 and have a Validity Period longer than 90 days.
+	// After 2019-08-01, clients MUST reject all certificates with this extension that have a Validity Period longer than 90 days.
+	may1st := time.Date(2019, time.May, 1,  0, 0, 0, 0, time.UTC)
```

Comment:
Move these to the top-level outside the function, and give them semantic names, e.g.:

```
// https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#cross-origin-cert-req
// Clients MUST reject certificates with this extension that were issued after 2019-05-01 and have a Validity Period longer than 90 days.
// After 2019-08-01, clients MUST reject all certificates with this extension that have a Validity Period longer than 90 days.
var start90DayGracePeriod = time.Date(2019, time.May, 1,  0, 0, 0, 0, time.UTC)
var end90DayGracePeriod = time.Date(2009, time.August, 1, 0, 0, 0, 0, time.UTC)
```

Reply:
fixed